library/network_connections.py

```diff
@@ -1133,6 +1135,28 @@ def connection_create(self, connections, idx, connection_current=None):
                 rr = NM.IPRoute.new(
                     r["family"], r["network"], r["prefix"], r["gateway"], r["metric"]
                 )
+                if isinstance(r["table"], int):
+                    NM.IPRoute.set_attribute(
+                        rr, "table", Util.GLib().Variant.new_uint32(r["table"])
+                    )
+                elif isinstance(r["table"], Util.STRING_TYPE):
+                    tableid = None
+                    with open("/etc/iproute2/rt_tables", "r") as f:
+                        for table in f.read().split("\n"):
+                            if (
+                                len(table.split()) >= 2
+                                and table.split()[1] == r["table"]
+                            ):
+                                tableid = int(table.split()[0])
+                    if tableid != None:
+                        NM.IPRoute.set_attribute(
+                            rr, "table", Util.GLib().Variant.new_uint32(tableid)
+                        )
```

Comment:
Parsing rt_tables should probably be a separate method and requires good unit test coverage.

Reply:
and, you wouldn't want to parse this more than once (e.g. for each tableid). Instead, parse it once and keep it around. The linux system role is not going to stick around for long, so it should handle that the file could change -- and if the file cannot change, there is no need to parse this repeatedly. Just cache it.

Reply:
Good idea, cache the mapping to avoid parsing again.

Reply:
I created the function `validate_route_tables()` for parsing the rt_tables, and I added the unit tests for testing the function.